### ISSUE TITLE
SubmitInput() fails with AMF_INVALID_RESOLUTION after calling Reinit() on BGRA surface

### ISSUE THREAD

**Issue by yujikosuga** [GPUOpen-LibrariesAndSDKs/AMF#216]
### Summary

When a surface is initialized with BGRA and `ReInit` is called for the purpose of resizing the surface, `SubmitInput` keeps failing with an error code `29`, which is `AMF_INVALID_RESOLUTION`. This issue occurs with BGRA, but not with NV12.

### Tested Environment

* Ubuntu 18.04
* amdgpu-pro 19.30-934563
* amf-amdgpu-pro 1.4.12-934563
* RX 480

### Steps to Reproduce

I created a test program to reproduce the issue at https://gist.github.com/yujikosuga/94dcd95a80023cb6b6040a009c52515a. This program first creates a surface of 1920x1080 resolution and passes it to `SubmitInput` for 10 times, then calls `Reinit` to resize the surface to 2020x1080 and `SubmitInput` for 20 times. The surface is created with BGRA if `BGRA_MODE` is defined, otherwise use NV12.

This is the output when I ran with BGRA. The result code before `Reinit` is `0` (`AMF_OK`) but `29` (`AMF_INVALID_RESOLUTION`) after `Reinit`. Even after using a larger value for `amf_sleep` (such as `amf_sleep(100)`), the result is the same.
```
0 result code: 0
1 result code: 0
2 result code: 0
3 result code: 0
4 result code: 0
5 result code: 0
6 result code: 0
7 result code: 0
8 result code: 0
9 result code: 0
resize from 1920x1080 to 2020x1080
10 result code: 29
11 result code: 29
12 result code: 29
13 result code: 29
14 result code: 29
15 result code: 29
16 result code: 29
17 result code: 29
18 result code: 29
19 result code: 29
20 result code: 29
21 result code: 29
22 result code: 29
23 result code: 29
24 result code: 29
25 result code: 29
26 result code: 29
27 result code: 29
28 result code: 29
29 result code: 29
```

And this is the output with NV12. The result code is always `0`, which is `AMF_OK`.
```
0 result code: 0
1 result code: 0
2 result code: 0
3 result code: 0
4 result code: 0
5 result code: 0
6 result code: 0
7 result code: 0
8 result code: 0
9 result code: 0
resize from 1920x1080 to 2020x1080
10 result code: 0
11 result code: 0
12 result code: 0
13 result code: 0
14 result code: 0
15 result code: 0
16 result code: 0
17 result code: 0
18 result code: 0
19 result code: 0
20 result code: 0
21 result code: 0
22 result code: 0
23 result code: 0
24 result code: 0
25 result code: 0
26 result code: 0
27 result code: 0
28 result code: 0
29 result code: 0
```

### Questions

1. Is the way that I use `ReInit` wrong? Do I need to call some other functions for BGRA?
2. Is there any quick workaround to avoid this issue when I use BGRA except for manually converting a surface from BGRA to NV12 before calling `SubmitInput`?

**Comment by MikhailAMD** [comment]
I should check tomorrow, but you can just call Terminate and Init as a way-around. It should not be much different from performance perspective .

**Comment by MikhailAMD** [comment]
Could you please try the latest driver?

**Comment by yujikosuga** [comment]
The issue still exists on AMDGPU-PRO 20.45 with AMF 1.4.18. The error code `29` was still printed for BGRA.

**Comment by MikhailAMD** [comment]
Yes, working on ReInit for Vulkan.

**Comment by MikhailAMD** [comment]
This should be fixed with the latest driver release (21.6.1)

**Comment by yujikosuga** [comment]
Thanks. I confirmed the issue was fixed on AMDGPU Pro 21.20-1271047 on Ubuntu 20.04. Let me close this ticket. Thanks again!
